src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -201,8 +201,7 @@ private DetailNode convertParseTreeToDetailNode(ParseTree parseTreeNode) {
         while (currentJavadocParent != null) {
             // remove unnecessary children tokens
             if (currentJavadocParent.getType() == JavadocTokenTypes.TEXT) {
-                currentJavadocParent
-                        .setChildren((DetailNode[]) JavadocNodeImpl.EMPTY_DETAIL_NODE_ARRAY);
```

Comment:
Redundant type cast

Reply:
I believe there was an eclipse setting for the same violation but we couldn't enable it because of TC. We should be able to enable it now.